apiserver/listener_test.go

```diff
@@ -157,14 +158,17 @@ func (s *listenerSuite) TestPause(c *gc.C) {
 	}()
 
 	start <- true
-	s.clock.Advance((l.minPause/time.Millisecond - 1) * time.Millisecond)
+	minPauseTimeGone := time.After(50 * time.Millisecond)
+	err := s.clock.WaitAdvance(time.Millisecond, (l.minPause/time.Millisecond-1)*time.Millisecond, 1)
+	c.Assert(err, jc.ErrorIsNil)
 	select {
 	case <-done:
 		c.Fatal("pause returned too soon")
-	case <-time.After(50 * time.Millisecond):
+	case <-minPauseTimeGone:
 	}
 	c.Assert(s.listener.count, gc.Equals, 0)
-	s.clock.Advance(time.Millisecond)
+	err = s.clock.WaitAdvance(10*time.Millisecond, time.Millisecond, 1)
+	c.Assert(err, jc.ErrorIsNil)
 	select {
 	case <-done:
 	case <-time.After(10 * time.Second):
```

Comment:
use coretesting.LongWait here? i.e. the amount of time to when something is expected to happen